keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/handlers/ManualJudgmentNotificationHandler.kt

```diff
@@ -50,6 +50,20 @@ class ManualJudgmentNotificationHandler(
           gitDataGenerator.generateScmInfo(this, application, artifactCandidate)
         }
 
+        // Add a warning section in case there's a pinned artifact
+        if (pinnedArtifact != null){
+          section {
+            markdownText(":warning: Another version is pinned here. You will need to unpin it first to promote this version.")
+            accessory {
+              button {
+                text("See pinned version", emoji = true)
+                actionId("button:url:pinned")
+                url(gitDataGenerator.generateArtifactUrl(application, pinnedArtifact.reference, pinnedArtifact.version))
```

Comment:
It looks like `generateArtifactUrl` is only used in this class and so should probably go here instead of in `GitDataGenerator`.

Reply:
all the relevant data to constructing the URL is there, and we will probably need it for future use :)

Reply:
and i also added some other usage for it